src/rgw/rgw_asio_frontend.cc

```diff
@@ -220,64 +226,104 @@ class AsioFrontend {
   void unpause(RGWRados* store, rgw_auth_registry_ptr_t);
 };
 
-int AsioFrontend::init()
+unsigned short parse_port(const char *input, boost::system::error_code& ec)
+{
+  char *end = nullptr;
+  auto port = std::strtoul(input, &end, 10);
+  if (port > std::numeric_limits<unsigned short>::max()) {
+    ec.assign(ERANGE, boost::system::system_category());
+  } else if (port == 0 && end == input) {
+    ec.assign(EINVAL, boost::system::system_category());
+  }
+  return port;
+}
+
+tcp::endpoint parse_endpoint(std::experimental::string_view input,
```

Comment:
Does Ceph have any policy on `std::experimental`? What about using `boost::string_view` instead?

Reply:
ah, thanks for reminding me about this part

i had found that `boost::asio::ip::make_address()` wouldn't accept `boost::string_view` or `std::string_view`, so this was the only form that would compile. but now that i look closer (http://www.boost.org/doc/libs/1_66_0/boost/asio/detail/string_view.hpp), there are typedefs for `string_view`/`basic_string_view` in namespace `boost::asio` that i can use instead